src/uct/ib/mlx5/dv/ib_mlx5dv_md.c

```diff
@@ -1179,6 +1264,209 @@ static void uct_ib_mlx5_devx_md_cleanup(uct_ib_md_t *ibmd)
     ucs_recursive_spinlock_destroy(&md->dbrec_lock);
 }
 
+#if HAVE_DECL_MLX5DV_DEVX_UMEM_REG_EX
+
+static ucs_status_t
+uct_ib_mlx5_devx_reg_exported_key(uct_ib_md_t *ib_md, uct_ib_mem_t *ib_memh)
+{
+    uct_ib_mlx5_md_t *md    = ucs_derived_of(ib_md, uct_ib_mlx5_md_t);
+    uct_ib_mlx5_mem_t *memh = ucs_derived_of(ib_memh, uct_ib_mlx5_mem_t);
+    char in[UCT_IB_MLX5DV_ST_SZ_BYTES(create_mkey_in)]                = {0};
+    char out[UCT_IB_MLX5DV_ST_SZ_BYTES(create_mkey_out)]              = {0};
+    struct mlx5dv_pd dvpd                                             = {0};
+    struct mlx5dv_obj dv                                              = {{0}};
+    char ein[UCT_IB_MLX5DV_ST_SZ_BYTES(allow_other_vhca_access_in)]   = {0};
+    char eout[UCT_IB_MLX5DV_ST_SZ_BYTES(allow_other_vhca_access_out)] = {0};
+    struct mlx5dv_devx_umem_in umem_in;
+    ucs_status_t status;
+    void *access_key;
+    void *address;
+    size_t length;
+    void *mkc;
+    int ret;
+
+    address = memh->mrs[UCT_IB_MR_DEFAULT].super.ib->addr;
+    length  = memh->mrs[UCT_IB_MR_DEFAULT].super.ib->length;
+
+    /* register umem */
+    umem_in.addr        = address;
+    umem_in.size        = length;
+    umem_in.access      = UCT_IB_MLX5_MD_UMEM_ACCESS;
+    umem_in.pgsz_bitmap = UINT64_MAX & ~(UCS_BIT(UCT_IB_MLX5_PAGE_SHIFT) - 1);
+    umem_in.comp_mask   = 0;
+
+    ucs_assert(memh->umem == NULL);
+    memh->umem = mlx5dv_devx_umem_reg_ex(md->super.dev.ibv_context, &umem_in);
+    if (memh->umem == NULL) {
+        uct_md_log_mem_reg_error(0, "mlx5dv_devx_umem_reg_ex() failed: %m");
```

Comment:
uct_md_log_mem_reg_error should accept md and print the device name

Reply:
added new macro-wrapper to get device name from IB MD and pass to `uct_md_log_mem_reg_error` macro